src/test/unit/specific_feature_tests/dialogs.js

```diff
@@ -0,0 +1,489 @@
+}
+
+test("Link dialog doesn't open when no selection; by toolbar button click",
+    function () {
+        var dialogName = "CreateLink";
+        dialogTestHelper({
+            currentTest: this,
+            noChangeHtml: "<p>Foo</p>",
+            prepareFunc: function (wymeditor) {
+                wymeditor.deselect();
+            },
+            dialogButtonSelector: getDialogToolbarSelector(dialogName),
+            expectedOpenedOrNot: dialogTestHelper.EXPECT_NOT,
+            expectedTitle: getDialogDocumentTitle(dialogName)
+        });
+    }
+);
+
+test("Link dialog opens when selection is non-collapsed and " +
+    "`selectedContainer` is a non-anchor element; by toolbar button click",
+    function () {
+        var dialogName = "CreateLink";
+        dialogTestHelper({
+            currentTest: this,
+            noChangeHtml: "<p>Foo</p>",
+            prepareFunc: function (wymeditor) {
+                var p = wymeditor.body().childNodes[0];
+                makeTextSelection(
+                    wymeditor,
+                    p,
+                    p,
+                    0,
+                    3
+                );
+            },
+            dialogButtonSelector: getDialogToolbarSelector(dialogName),
+            expectedOpenedOrNot: dialogTestHelper.EXPECT_OPENED,
+            expectedTitle: getDialogDocumentTitle(dialogName)
+        });
+    }
+);
+
+test(
+    "Link dialog doesn't open when selection is collapsed and " +
+    "`selectedContainer` doesn't return false; by toolbar button click",
+    function () {
+        var dialogName = "CreateLink";
+        dialogTestHelper({
+            currentTest: this,
+            noChangeHtml: "<p>Foo</p>",
+            setCaretInSelector: "p",
+            dialogButtonSelector: getDialogToolbarSelector(dialogName),
+            expectedOpenedOrNot: dialogTestHelper.EXPECT_NOT,
+            expectedTitle: getDialogDocumentTitle(dialogName)
+        });
+    }
+);
+
+test("Link dialog opens when selection is collapsed and " +
+    "`selectedContainer` returns an `a` element; by toolbar button click",
+    function () {
+        var dialogName = "CreateLink";
+        dialogTestHelper({
+            currentTest: this,
+            noChangeHtml: "<p><a href=\"http://example.com/\">Foo</a></p>",
+            prepareFunc: function (wymeditor) {
+                var a = wymeditor.$body().find("a")[0];
+                makeTextSelection(
+                    wymeditor,
+                    a,
+                    a,
+                    1,
+                    1
+                );
+            },
+            dialogButtonSelector: getDialogToolbarSelector(dialogName),
+            expectedOpenedOrNot: dialogTestHelper.EXPECT_OPENED,
+            expectedTitle: getDialogDocumentTitle(dialogName)
+        });
+    }
+);
+
+test("Link dialog opens when selection is non-collapsed and " +
+    "`selectedContainer` returns an `a` element; by toolbar button click",
```

Comment:
> What about a collapsed selection within an anchor tag?

@winhamwr I think you meant "non–collapsed". Here, I added another one for a non–collapsed in an anchor.